projects/cashmere/src/lib/stepper/stepper.component.scss

```diff
@@ -0,0 +1,324 @@
+$red-complete: lighten(tint($red, 20%), 3%);
+
+$track-bg: $slate-gray-100;
+$track-bg-iso: $slate-gray-200;
+$border: $slate-gray-200;
+$disk-bg: $white;
+
+.hc-step {
+    @include hc-step();
+
+    .hc-stepper-arrow & {
+        @include hc-stepper-arrow-step();
+        @include hc-stepper-border-top($track-bg);
+    }
+    .hc-stepper-isolated & {
+        @include hc-stepper-isolated-step($track-bg-iso);
+    }
+}
+
+.hc-step:not(.hc-step-active):not(.hc-step-last):not(.hc-step-completed) {
+    .hc-stepper-arrow & {
+        @include hc-stepper-arrow-after($border, $border);
+    }
+}
+
+.hc-step-disk {
+    .hc-stepper-arrow & {
+        @include hc-stepper-arrow-step-disk();
+        @include hc-disk-border($border);
+        @include hc-stepper-bg-color($disk-bg);
+    }
+    .hc-stepper-isolated & {
+        @include hc-stepper-isolated-step-disk();
+        @include hc-disk-border($track-bg);
+        @include hc-stepper-bg-color($disk-bg);
+    }
+    .hc-stepper-green .hc-step-active &,
+    .hc-stepper-green .hc-step-complete &,
+    .hc-stepper-green .hc-step-completed & {
+        @include hc-disk-border($green);
+    }
+    .hc-stepper-blue .hc-step-active &,
+    .hc-stepper-blue .hc-step-complete &,
+    .hc-stepper-blue .hc-step-completed & {
+        @include hc-disk-border($blue);
+    }
+    .hc-stepper-purple .hc-step-active &,
+    .hc-stepper-purple .hc-step-complete &,
+    .hc-stepper-purple .hc-step-completed & {
+        @include hc-disk-border($purple);
+    }
+    .hc-stepper-orange .hc-step-active &,
+    .hc-stepper-orange .hc-step-complete &,
+    .hc-stepper-orange .hc-step-completed & {
+        @include hc-disk-border($orange);
+    }
+    .hc-stepper-red .hc-step-active &,
+    .hc-stepper-red .hc-step-complete &,
+    .hc-stepper-red .hc-step-completed & {
+        @include hc-disk-border($red);
+    }
+    .hc-stepper-green .hc-stepper-arrow .hc-step-complete &,
+    .hc-stepper-green .hc-stepper-arrow .hc-step-completed & {
+        @include hc-disk-border($green-complete);
+    }
+    .hc-stepper-blue .hc-stepper-arrow .hc-step-complete &,
+    .hc-stepper-blue .hc-stepper-arrow .hc-step-completed & {
+        @include hc-disk-border($blue-complete);
+    }
+    .hc-stepper-purple .hc-stepper-arrow .hc-step-complete &,
+    .hc-stepper-purple .hc-stepper-arrow .hc-step-completed & {
+        @include hc-disk-border($purple-complete);
+    }
+    .hc-stepper-orange .hc-stepper-arrow .hc-step-complete &,
+    .hc-stepper-orange .hc-stepper-arrow .hc-step-completed & {
+        @include hc-disk-border($orange-complete);
+    }
+    .hc-stepper-red .hc-stepper-arrow .hc-step-complete &,
+    .hc-stepper-red .hc-stepper-arrow .hc-step-completed & {
+        @include hc-disk-border($red-complete);
+    }
+}
+
+.hc-step-arrow {
+    .hc-stepper-arrow .hc-step:not(.hc-step-last) & {
+        @include hc-step-arrow($track-bg, null);
+    }
+    .hc-stepper-green .hc-stepper-arrow .hc-step-completed & {
```

Comment:
We're getting some pretty high specificity here. Can some of these intermediate selectors be omitted? For example, in `.hc-stepper-green .hc-stepper-arrow .hc-step-completed .hc-step-arrow`, could we safely remove the `.hc-stepper-arrow`? Maybe we could further condense things by adding a conditional class e.g. `hc-step-arrow-completed` in the markup so we wouldn't need the `.hc-step-completed` class. I think the extra effort is worth it to make Cashmere styles as overrideable as possible.

Reply:
In fact, throughout this file I think it would be good to achieve a specificity goal of <3.

Reply:
yep - `hc-step-arrow-completed` could work

Reply:
@isaaclyman I've pushed a commit the implements this suggestion